cloud/scope/cluster.go

```diff
@@ -140,6 +140,15 @@ func (s *ClusterScope) Close() error {
 	return s.patchHelper.Patch(context.TODO(), s.AzureCluster)
 }
 
+// AdditionalTags returns AdditionalTags from the scope's AzureCluster. The returned value will never be nil.
+func (s *ClusterScope) AdditionalTags() infrav1.Tags {
+	if s.AzureCluster.Spec.AdditionalTags == nil {
+		s.AzureCluster.Spec.AdditionalTags = infrav1.Tags{}
```

Comment:
I'm not fond of mutating the object inside getters. it is unexpected, and you return a copy of it anyway.
If we want to avoid it being nil, we could initialize it earlier in the stack.

Reply:
+1

Reply:
fixed, thanks for catching